VMAccess/vmaccess.py

```diff
@@ -157,24 +208,33 @@ def _remove_user_account(user_name, hutil):
 
     try:
         sudoers = _get_other_sudoers(user_name)
-        waagent.MyDistro.DeleteAccount(user_name)
+        MyDistro.delete_account(user_name)
         _save_other_sudoers(sudoers)
     except Exception as e:
-        waagent.AddExtensionEvent(name=hutil.get_name(),
-                                  op=waagent.WALAEventOperation.Enable,
-                                  isSuccess=False,
-                                  message="(02102)Failed to remove user.")
+        ext_utils.add_extension_event(name=hutil.get_name(),
+                                      op=constants.WALAEventOperation.Enable,
+                                      is_success=False,
+                                      message="(02102)Failed to remove user.")
         raise Exception("Failed to remove user {0}".format(e))
 
-    waagent.AddExtensionEvent(name=hutil.get_name(),
-                              op=waagent.WALAEventOperation.Enable,
-                              isSuccess=True,
-                              message="Successfully removed user")
+    ext_utils.add_extension_event(name=hutil.get_name(),
+                                  op=constants.WALAEventOperation.Enable,
+                                  is_success=True,
+                                  message="Successfully removed user")
 
 
 def _set_user_account_pub_key(protect_settings, hutil):
-    ovf_xml = waagent.GetFileContents('/var/lib/waagent/ovf-env.xml')
-    ovf_env = waagent.OvfEnv().Parse(ovf_xml)
+    ovf_xml = None
+    ovf_env = None
+    try:
+        ovf_xml = ext_utils.get_file_contents('/var/lib/waagent/ovf-env.xml')
+        ovf_env = ovf_utils.OvfEnv.parse(ovf_xml, Configuration)
+    except (OSError, ValueError, KeyError, AttributeError):
+        pass
+    if ovf_xml is None or ovf_env is None:
```

Comment:
It should be sufficient to check if ovf_env is None.

Reply:
Being extra cautions doesn't hurt this case.